p2p/message.go

```diff
@@ -0,0 +1,76 @@
+package p2p
+
+import (
+	"bytes"
+
+	xdr "github.com/stellar/go-xdr/xdr3"
+)
+
+type Message struct {
+	V  int32
+	V0 MessageV0
+}
+
+func (m *Message) MarshalBinary() ([]byte, error) {
+	b := bytes.Buffer{}
+	_, err := xdr.Marshal(&b, m)
+	if err != nil {
+		return nil, err
+	}
+	return b.Bytes(), nil
+}
+
+func (m *Message) UnmarshalBinary(b []byte) error {
+	_, err := xdr.Unmarshal(bytes.NewReader(b), m)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type MessageV0 struct {
+	Chain      Chain
+	Body       []byte
+	Signatures [][]byte
+}
+
+func (m *MessageV0) MarshalBinary() ([]byte, error) {
+	b := bytes.Buffer{}
+	_, err := xdr.Marshal(&b, m)
+	if err != nil {
+		return nil, err
+	}
+	return b.Bytes(), nil
+}
+
+func (m *MessageV0) UnmarshalBinary(b []byte) error {
+	_, err := xdr.Unmarshal(bytes.NewReader(b), m)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type Chain int32
+
+func (c Chain) MarshalBinary() ([]byte, error) {
+	b := bytes.Buffer{}
+	_, err := xdr.Marshal(&b, &c)
+	if err != nil {
+		return nil, err
+	}
+	return b.Bytes(), nil
+}
+
+func (c *Chain) UnmarshalBinary(b []byte) error {
+	_, err := xdr.Unmarshal(bytes.NewReader(b), c)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+const (
+	ChainStellar  Chain = 0
```

Comment:
we should reuse this from chain.go
Or if anything we should have these of chainIDs in a common place and chain.go can reuse that
what do you think?

Reply:
Yes agreed. I'll follow up with that change.